cmd/core-metadata/main.go

```diff
@@ -48,16 +51,25 @@ func main() {
 	flag.Usage = usage.HelpCallback
 	flag.Parse()
 
-	httpServer := httpserver.NewBootstrap(metadata.LoadRestRoutes())
+	configuration := &metadataConfig.ConfigurationStruct{}
+	dic := di.NewContainer(di.ServiceConstructorMap{
+		container.ConfigurationName: func(get di.Get) interface{} {
+			return configuration
+		},
+		bootstrapContainer.ConfigurationInterfaceName: func(get di.Get) interface{} {
+			return get(container.ConfigurationName)
+		},
```

Comment:
At this point, your container should be instantiated with an empty map.  

When you tackle the follow-on issue to eliminate the Configuration global, you'll need to add these.  And when you do, you'll need to spin your own helper to type assert and return metadata's configuration struct (you're using system management agent's helper and returning the system management agent's configuration struct in the above code).